### PR TITLE
Clean up notebookrenderer and finish its tests

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: test/src/notebook/notebook/widget.spec.ts [hunk 1]
```diff
@@ -73,11 +133,248 @@ describe('notebook/notebook/widget', () => {
...
+        expect(() => { widget.rendermime = null; }).to.throwError();
+      });
+
+    });
+
+    describe('#childAt()', () => {
+
+      it('should get the child widget at a specified index', (done) => {
+        let widget = createWidget();
+        widget.attach(document.body);
```
Comment (jasongrout) [inline test/src/notebook/notebook/widget.spec.ts]: The widget.attach tests are probably brittle - there's no guarantee that our requestAnimationFrame will happen after the message loop's, is there? That said, they probably will work fine.

CC @sccolbert.

Comment (blink1073) [inline test/src/notebook/notebook/widget.spec.ts]: The requests are handled in order, and this is made after the request from update message.

Comment (blink1073) [inline test/src/notebook/notebook/widget.spec.ts]: We use this extensively in the `phosphor` test suite.

Comment (jasongrout) [inline test/src/notebook/notebook/widget.spec.ts]: Reference: https://www.w3.org/TR/html51/webappapis.html#animation-frames implies the order is preserved. Thanks!

**Pull request merged**